src/neuroconv/tools/nwb_helpers/_models/_hdf5_dataset_models.py

```diff
@@ -0,0 +1,76 @@
+"""Base Pydantic models for the HDF5DatasetConfiguration."""
+from typing import Any, Dict, Literal, Union
+
+import h5py
+from nwbinspector.utils import is_module_installed
+from pydantic import Field
+
+from ._base_dataset_models import DatasetConfiguration
+
+_base_hdf5_filters = set(h5py.filters.decode) - set(
+    (
+        "shuffle",  # controlled via H5DataIO
+        "fletcher32",  # controlled via H5DataIO
+        "scaleoffset",  # enforced indrectly by HDMF/PyNWB data types
+    )
+)
+_available_hdf5_filters = set(_base_hdf5_filters)
+if is_module_installed(module_name="hdf5plugin"):
+    import hdf5plugin
+
+    _available_hdf5_filters = _available_hdf5_filters | set(
+        (str(hdf5plugin_filter).rstrip("'>").split(".")[-1] for hdf5plugin_filter in hdf5plugin.get_filters())
+    )  # Manual string parsing because of slight mismatches between .filter_name and actual import class
+AVAILABLE_HDF5_COMPRESSION_METHODS = tuple(_available_hdf5_filters)
+
+
+class HDF5DatasetConfiguration(DatasetConfiguration):
+    """A data model for configuring options about an object that will become a HDF5 Dataset in the file."""
+
+    # TODO: When using Pydantic v2, replace with `model_config = ConfigDict(...)`
+    class Config:
+        arbitrary_types_allowed = True
+        validate_assignment = True
+
+    compression_method: Union[
+        Literal[AVAILABLE_HDF5_COMPRESSION_METHODS], h5py._hl.filters.FilterRefBase, None
+    ] = Field(
+        default="gzip",
+        description=(
+            "The specified compression method to apply to this dataset. "
+            "Can be either a string that matches an available method on your system, "
+            "or an instantiated h5py/hdf5plugin object."
+            "Set to `None` to disable compression."
+        ),
+    )
+    # TODO: actually provide better schematic rendering of options. Only support defaults in GUIDE for now.
+    # Looks like they'll have to be hand-typed however... Can try parsing the google docstrings - no annotation typing.
+    compression_options: Union[Dict[str, Any], None] = Field(
+        default=None, description="The optional parameters to use for the specified compression method."
+    )
+
+    def get_data_io_keyword_arguments(self) -> Dict[str, Any]:
```

Comment:
```suggestion
    def get_data_io_keyword_arguments(self) -> Dict[str, Any]:
    """
    You can use like this
    io = H5DataIO(**get_data_io_keyword_arguments())
    """
```
Can you add some clarification to the docstring like this. Also for Zarr.

This is one of the few instances that I am tempted to use `_kwargs` instead of `keyword_arguments`.

Reply:
I see that is meant to be used with the `set_data_io` container method. I need to think more about this. I also read:

data_io_kwargs = dataset_configuration.get_data_io_kwargs()` in another PR. But could not locate that function.

Reply:
https://github.com/catalystneuro/neuroconv/pull/571/files#diff-91679913ea0f31861b6d5e7275fa4a7e03b256d6cd91aa8d82ca505c604d9112R226

Reply:
Thanks, just to clarify. Do you change the name in another PR further down the chain from `get_data_io_keyword_arguments` to `get_data_io_kwargs()` ?

Reply:
Looks like it is still the more verbose version: https://github.com/catalystneuro/neuroconv/blob/integrate_with_interfaces_and_converter/src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py#L53

Reply:
I prefer the more verbose version to help external readers understand it faster